pyanaconda/modules/payload/live/initialization.py

```diff
@@ -156,6 +156,55 @@ def run(self):
         return size
 
 
+class DownloadProgress(object):
+    """Provide methods for download progress reporting."""
+
+    def __init__(self, task):
+        """Create a progress object for given task.
+
+        :param task: task to report in
+        :type task: Task
+        """
+        self._task = task
+
+    def start(self, url, size):
+        """Start of download
+
+        :param url: url of the download
+        :type url: str
+        :param size: length of the file
+        :type size: int
+        """
+        self.url = url
+        self.size = size
+        self._pct = -1
+
+    def update(self, bytes_read):
+        """Download update
+
+        :param bytes_read: Bytes read so far
+        :type bytes_read:  int
+        """
+        if not bytes_read:
+            return
+        pct = min(100, int(100 * bytes_read / self.size))
+
+        if pct == self._pct:
+            return
+        self._pct = pct
+        self._task.report_progress("Downloading image %(url)s (%(pct)d%%)" %
+                                   {"url": self.url, "pct": pct})
+
+    def end(self, bytes_read):
+        """Download complete
```

Comment:
You are missing period in the doc strings here.

Reply:
Fixed